src/api_proxy/service_control/request_info.h

```diff
@@ -141,18 +141,34 @@ enum ScResponseErrorType {
   CONSUMER_QUOTA = 5,
 };
 
+// Store the info of service control http call failure/call error response.
+struct ScResponseError {
+  std::string name;
+
+  bool is_error_from_http_call;
+
+  ScResponseErrorType type;
+};
+
 // Stores the information extracted from the check response.
 struct CheckResponseInfo {
-  ScResponseErrorType error_type = ScResponseErrorType::ERROR_TYPE_UNSPECIFIED;
-
   std::string consumer_project_number;
 
   std::string consumer_type;
 
   std::string consumer_number;
 
+  ScResponseError error;
+
   // The trust level of the API Key that was checked.
-  api_key::ApiKeyState api_key_state = api_key::ApiKeyState::NOT_CHECKED;
+  api_key::ApiKeyState api_key_state;
+
+  CheckResponseInfo()
+      : consumer_project_number(),
+        consumer_type(),
+        consumer_number(),
+        error({"", false, ScResponseErrorType::ERROR_TYPE_UNSPECIFIED}),
```

Comment:
can we define a default constructor for ScResponseError? move these default values there